### PR TITLE
Separate external network from Kubernetes quoin

### PR DESCRIPTION
**Pull request by nicklathe**
- Abstract external VPC creation out of Kubernetes into it's own quoin
    - Update docs.md

### CODE REVIEW

**Comment by nicklathe** [comment]
In taking another pass, I'm realizing this PR is not complete yet. Will continue to work through it. Feel free to post feedback on things I'm missing.

**Comment by dragan** [comment]
Close, what we want to create is an internal layout only quoin. You can use the external-internal layout as a template and then remove the external bits, like the Internet Gateway and NAT Gateways.

**Comment by dragan** [comment]
Hmm, but I like the approach though and see what you're trying to do. We need to think about the tag though so the ELB's that K8S create get put on the correct subnet. Which is partly why I thought it would be rip out the network piece all together and not do the layout at all inside the K8S quoin.

**Comment by nicklathe** [comment]
Oh I see now. I had misunderstood. I assumed that because Kubernetes needed an internal network layout, it would be included in the Kubernetes quoin, and we would abstract the optional external network piece out into a new quoin. However, I do like separating the network component altogether away from Kubernetes. I'll work on some changes, and update this PR.

**Comment by nicklathe** [comment]
Ok, PR updated based on feedback. More feedback welcome. Once this looks good, I will also open a PR into our `infra` repo, and make the appropriate changes there.

**Comment by nicklathe** [comment]
@dragan @weitenghuang I worked this afternoon trying to test this locally by creating a k8 cluster in  our integration VPC, but was running into errors. I'm sure I was doing something wrong somewhere, but I'm not sure what (I was just trying to build a k8 cluster on existing infra layers that are already there).

**Comment by dragan** [comment]
@nicklathe No worries, if you could, include the error next time in your comment. If you could, show me Tuesday morning. @weitenghuang, if get a chance tomorrow, try testing this PR and providing feedback. However, no worries if you can't. We'll look at it together on Tuesday. Have a great weekend!

**Comment by nicklathe** [comment]
@dragan @weitenghuang this should be ready for a merge. Wei and I tested it yesterday, and other than the fact that building a k8 cluster with internal subnets only (no NATs) in our us-west-2 account couldn't reach out to quay, it appears the changes I made work. I will test this morning to create a k8 cluster using the external-internal-layout to finish confirming this PR works.

**Comment by nicklathe** [comment]
Ok, updated `docs.md`. @weitenghuang, let me know if I missed something.

File: docs.md [hunk 1]
```diff
@@ -383,6 +369,7 @@ provider "aws" {
 | availability_zones | Comma separated list of availability zones for a region. | - | yes |
 | external_subnets | Comma separated list of CIDR's to use for the external subnets. | - | yes |
 | internal_subnets | Comma separated list of CIDR's to use for the internal subnets. | - | yes |
```
Comment (weitenghuang) [inline docs.md]: Above two rows should be removed, and add new required variables

**Pull request merged**